frontend/src/components/tables/DataTablePagination.tsx

```diff
@@ -0,0 +1,140 @@
+import { ArrowLeftIcon, ChevronLeftIcon, ChevronRightIcon, ArrowRightIcon } from '@chakra-ui/icons';
+import {
+  Flex,
+  Tooltip,
+  IconButton,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  NumberIncrementStepper,
+  NumberDecrementStepper,
+  Select,
+  Text,
+} from '@chakra-ui/react';
+import { type Table } from '@tanstack/react-table';
+import React from 'react';
+
+interface DataTablePaginationProps<T> {
+  table: Table<T>;
+}
+
+const numEntriesPerPageOptions = [10, 20, 30, 40, 50];
+
+const DataTablePagination = <T extends object>({ table }: DataTablePaginationProps<T>): JSX.Element => {
+  const {
+    getPageCount,
+    getCanPreviousPage,
+    getCanNextPage,
+    nextPage,
+    previousPage,
+    setPageIndex,
+    setPageSize,
+    getState,
+  } = table;
+  const { pageIndex, pageSize } = getState().pagination;
+  return (
+    <Flex justifyContent="space-between" margin={4} alignItems="center">
+      <Flex>
+        {/* Jump to first page icon button */}
+        <Tooltip label="First Page">
+          <IconButton
+            aria-label="First Page"
+            onClick={() => {
+              setPageIndex(0);
+            }}
+            isDisabled={!getCanPreviousPage()}
+            icon={<ArrowLeftIcon height={3} width={3} />}
+            mr={4}
+          />
+        </Tooltip>
+
+        {/* Navigate to previous page icon button */}
+        <Tooltip label="Previous Page">
+          <IconButton
+            aria-label="Previous Page"
+            onClick={previousPage}
+            isDisabled={!getCanNextPage()}
+            icon={<ChevronLeftIcon height={6} width={6} />}
+          />
+        </Tooltip>
+      </Flex>
+
+      <Flex alignItems="center">
+        {/* Current page number display */}
+        <Text fontSize="sm" flexShrink="0" marginRight={8}>
+          {'Page '}
+          <Text fontSize="sm" fontWeight="bold" as="span">
+            {pageIndex + 1}
+          </Text>
+          {' of '}
+          <Text fontSize="sm" fontWeight="bold" as="span">
+            {getPageCount()}
+          </Text>
+        </Text>
+
+        {/* Page Navigation Field Input */}
+        <Text fontSize="sm" flexShrink="0">
+          {'Go to page: '}
+        </Text>
+        <NumberInput
+          size="sm"
+          marginLeft={2}
+          marginRight={8}
+          width={28}
+          min={1}
+          max={getPageCount()}
+          onChange={(_str, num) => {
+            if (_str !== '') setPageIndex(Math.max(num - 1, 0));
```

Comment:
```suggestion
            if (_str !== '') setPageIndex(Math.min(Math.max(num - 1, 0), getPageCount() - 1));
```

still missing a page cap (test with 3)
(i edited the previous comment before so i think it didn't get added to the commit, my bad)

Reply:
Thanks for this!